asyncpg/connection.py

```diff
@@ -400,47 +426,124 @@ async def _introspect_types(self, typeoids, timeout):
         return await self.__execute(
             self._intro_query, (list(typeoids),), 0, timeout)
 
-    def cursor(self, query, *args, prefetch=None, timeout=None):
+    def cursor(
+        self,
+        query,
+        *args,
+        prefetch=None,
+        timeout=None,
+        record_class=None
```

Comment:
Instead of adding `record_class` everywhere I'd allow passing it to the constructor. Or add a `set_record_class_factory` method.

Reply:
Is there any evidence that users want multiple different record types on one connection?

Reply:
The entire impetus of this push was from #577, where the `record_class` is used as a `TypeDict`-like typehint for the result, which gets typechecked also.

> I'd allow passing it to the constructor. 

It is also allowed to be passed in the constructor if you want a connection-wide subclass.